101-application-gateway-rewrite/azuredeploy.json

```diff
@@ -0,0 +1,380 @@
+        }
+      ],
+      "metadata": {
+        "description": "An array of json objects like this : {'name':name, 'value':value}"
+      }
+    }
+  },
+  "variables": {
+    "applicationGatewayName": "[concat(resourceGroup().name, '-appgw')]",
+    "idName": "identity1",
+    "keyVaultName": "yourvault-kv",
+    "publicIPAddressName": "[concat(resourceGroup().name, '-publicIp')]",
+    "virtualNetworkName": "[concat(resourceGroup().name, '-vnet')]",
+    "publicIPRef": "[resourceId('Microsoft.Network/publicIPAddresses',variables('publicIPAddressName'))]",
+    "identityID": "[resourceId('Microsoft.ManagedIdentity/userAssignedIdentities',variables('idName'))]",
+    "keyVaultID": "[resourceId('Microsoft.KeyVault/vaults', variables('keyVaultName'))]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.ManagedIdentity/userAssignedIdentities",
+      "apiVersion": "2015-08-31-PREVIEW",
+      "name": "[variables('idName')]",
+      "location": "[resourceGroup().location]"
+    },
+    {
+      "type": "Microsoft.KeyVault/vaults",
+      "apiVersion": "['2018-02-14']",
+      "name": "[variables('keyVaultName')]",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[concat('Microsoft.ManagedIdentity/userAssignedIdentities/', variables('idName'))]"
+      ],
+      "properties": {
+        "sku": {
+          "name": "standard",
+          "family": "A"
+        },
+        "tenantId": "[reference(variables('identityID'), '2015-08-31-PREVIEW').tenantId]",
+        "accessPolicies": [
+          {
+            "tenantId": "[reference(variables('identityID'), '2015-08-31-PREVIEW').tenantId]",
+            "objectId": "[reference(variables('identityID'), '2015-08-31-PREVIEW').principalId]",
+            "permissions": {
+              "secrets": [
+                "get"
+              ]
+            }
+          }
+        ],
+        "enableSoftDelete": true
+      }
+    },
+    {
+      "type": "Microsoft.KeyVault/vaults/secrets",
+      "apiVersion": "['2018-02-14']",
+      "name": "[concat(variables('keyVaultName'), '/', parameters('Secrets')[copyIndex()].name)]",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[resourceId('Microsoft.KeyVault/vaults', variables('keyVaultName'))]"
+      ],
+      "copy": {
+        "name": "secretcopy",
+        "count": "[length(parameters('Secrets'))]"
+      },
+      "properties": {
+        "value": "[parameters('Secrets')[copyIndex()].value]",
+        "recoveryLevel": "Purgeable"
+      }
+    },
+    {
+      "type": "Microsoft.Network/publicIPAddresses",
+      "apiVersion": "['2018-10-01']",
+      "name": "[variables('publicIPAddressName')]",
+      "location": "[resourceGroup().location]",
+      "sku": {
+        "name": "Standard"
+      },
+      "properties": {
+        "publicIPAllocationMethod": "Static",
+        "dnsSettings": {
+            "domainNameLabel": "[resourceGroup().name]"
```

Comment:
this is not globally unique, generate a value using uniqueString() or just remove the dnsSettings property

Reply:
using uniqueString() now.

Reply:
tried removing only dnsSettings property but travis CI was failing. Tried uniqueString as well.
FInally removed the block.

Reply:
using uniqueString() now.